### ISSUE TITLE
How to require the js from node_modules

### ISSUE THREAD

**Issue by dannystaff** [mymth/vanillajs-datepicker#1]
`Module not found: Error: Can't resolve 'vanillajs-datepicker'`

Typically, when using npm packages; I can simply require them via their namespace. No matter what I try with this project, it cannot find the module. I've also tried the path supplied in the docs. Its a Laravel 6/Bootstrap 4 project. Compiling using Mix.

Any idea what I am doing wrong?

**Comment by mymth** [comment]
You can't `require()` the package nor the script files because the scripts are not written as CommonJS modules.

Since they are ES6 modules, you need to `import` them.

**Comment by dannystaff** [comment]
I tried using the import that you showed in the docs, but npm still threw the same 'not found' error. I'll have a play around with the pathing to see if its something to do with how im mapping to node_modules.

**Comment by mymth** [comment]
I guess https://webpack.js.org/concepts/module-resolution/#module-paths would be helpful.
Sorry if you already know about it or I misunderstand something.

**Comment by benface** [comment]
I'm experiencing the same issue. I'm using Laravel Mix as well (a wrapper for Webpack), and this line:

```
import Datepicker from 'vanillajs-datepicker';
```

throws this error:

```
ERROR in ./src/js/pages/checkout.js
Module not found: Error: Can't resolve 'vanillajs-datepicker' in '/Users/benface/Desktop/Dev/project/src/js/pages'
 @ ./src/js/pages/checkout.js 2:0-46 12:23-33
 @ ./src/js/app.js
```

**Comment by benface** [comment]
@mymth I found the cause: [`main` is set to an empty string in `package.json`](https://github.com/mymth/vanillajs-datepicker/blob/352c426c8ec0459c23de08646f0c1fc191b7f6f4/package.json#L5). If I change it to `js/Datepicker.js`, it finds it and compiles properly. Would you like me to submit a PR?

**Comment by mymth** [comment]
@benface, I intentionally didn't set the main field because the main file to export both Datepicker and DateRangePicker modules hasn't been created yet. And that's because I...

- wasn't sure if it's safe to use for ES module.
- couldn't decide whether to include a CommonJS module (I'm still reluctant to do so.) and how the final file layout should be.
- thought loading with path is more common for ES modules and thus, starting without it would be OK. (Apparently this assumption was wrong.)

After reading your comment, I took a look at how other packages do with the main field and learned that using it for ES module seems not really a problem. But it still looks like an improper use of the field.

With the next update, I'm going to add the main js file and it's going be set to the `module` field instead of `main`. The main field will remain unset unless I change my mind to include CommonJS module.

The additional usage will be like below.

```javascript
import { Datepicker, DateRangePicker } from 'vanillajs-datepicker';
```

So, I'd like to recommend you to change your code to import the module using the path, or your code will break after the update.

**Comment by benface** [comment]
Thanks for the explanation @mymth! I don't know much about the difference between ES modules and CommonJS modules, but after reading a bit on it, you're right that `module` seems to be the better choice. And it makes sense to export both `Datepicker` and `DateRangePicker` like that. Do you have an ETA for the next update?

**Comment by mymth** [comment]
No, not yet. But it should be at some point this month.

**Comment by mymth** [comment]
I'll close this as the feature discussed here is now available with v1.0.1 release.
